src/journal-remote/journal-remote.c

```diff
@@ -777,21 +795,46 @@ static int setup_microhttpd_socket(RemoteServer *s,
         return setup_microhttpd_server(s, fd, key, cert, trust);
 }
 
+static int null_timer_event_handler(sd_event_source *timer_event,
+                                    uint64_t usec,
+                                    void *userdata) {
+        return dispatch_http_event(timer_event, 0, 0, userdata);
+}
+
 static int dispatch_http_event(sd_event_source *event,
                                int fd,
                                uint32_t revents,
                                void *userdata) {
         MHDDaemonWrapper *d = userdata;
         int r;
+        fd_set rd, w, e;
+        MHD_UNSIGNED_LONG_LONG timeout = ULONG_LONG_MAX;
 
         assert(d);
 
-        r = MHD_run(d->daemon);
+        r = MHD_get_fdset(d->daemon, &rd, &w, &e, 0);
         if (r == MHD_NO) {
-                log_error("MHD_run failed!");
+                log_error("MHD_get_fdset failed!");
+                return -EINVAL;
+        }
+        r = MHD_run_from_select(d->daemon, &rd, &w, &e);
```

Comment:
Why do you use `MHD_run_from_select()` if you did not call `select()`?

Reply:
The select/epoll is done by sd_event_run.

Reply:
I'm talking about code here: it is useless to get fdset from MHD and then use `MHD_run_from_select()` **without calling select() on fdsets**.
You should just use `MHD_run()`.
Correct code (those lines) was changed by this patch to wrong code.

Reply:
You could ping me by mentioning (@Karlson2k) for review, if you wish.